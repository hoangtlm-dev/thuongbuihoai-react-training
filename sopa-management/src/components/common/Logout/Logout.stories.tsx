import type { Meta, StoryObj } from '@storybook/react';
import Logout from '.';

const meta: Meta<typeof Logout> = {
  component: Logout,
};

export default meta;
type Story = StoryObj<typeof Logout>;

export const DropdownSize: Story = {
  args: {}
};
