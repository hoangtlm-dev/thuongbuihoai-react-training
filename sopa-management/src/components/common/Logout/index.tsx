import {
  memo,
  useCallback,
  useState
} from 'react';
import { useNavigate } from 'react-router-dom';

// Components
import Button, { ButtonType } from '@common/Button';
import Dropdown from '@common/Dropdown';

// Public
import { User } from '@assets/icons';

// Helpers
import { clearStorage } from '@helpers/storage';

// Constants
import { STORAGE_KEY } from '@constants/common';

interface LogoutProps {
  className?: string;
}

const Logout = ({ className = '' }: LogoutProps) => {
  const navigate = useNavigate()
  const [showDropdown, setShowDropdown] = useState<boolean>(false);

  const handleToggle = useCallback(() => {
    setShowDropdown(!showDropdown);
  }, [showDropdown]);

  const handleLogout = () => {
    clearStorage(STORAGE_KEY.TOKEN);
    navigate('/login');
  }

  const handleToggleItem = useCallback(() => {
    handleToggle();
    handleLogout();
  }, [showDropdown]);

  return (
    <div className={`${className} account`} data-testid='logout'>
      <Button
        ariaLabel='User'
        children={<User />}
        type={ButtonType.btnIconPrimary}
        onClick={handleToggle}
      />
      {showDropdown && (
        <Dropdown
          isHref
          className='account-dropdown'
          onClick={handleToggleItem}
        />
      )}
    </div>
  );
};

export default memo(Logout);
