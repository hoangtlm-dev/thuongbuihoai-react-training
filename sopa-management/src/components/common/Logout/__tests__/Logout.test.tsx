import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import Logout from '..';

describe('Logout component', () => {
  test('should render Logout component', () => {
    const { getByTestId } = render(<Logout />);
    expect(getByTestId('logout')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<Logout />);
    expect(asFragment()).toMatchSnapshot();
  });
});
