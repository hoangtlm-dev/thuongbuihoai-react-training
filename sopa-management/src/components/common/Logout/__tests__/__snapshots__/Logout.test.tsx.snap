// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout component matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="logout"
  >
    <button
      class="btn btn-icon-primary btn-icon-primary"
      data-testid="button"
    >
      <div
        class="icon icon-user icon-user"
        data-testid="icon"
      />
    </button>
  </div>
</DocumentFragment>
`;
